subcommand/health-sync/command_test.go

```diff
@@ -0,0 +1,261 @@
+package healthsync
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestRunWithoutContainerNames(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+	code := cmd.Run(nil)
+	require.Equal(t, 0, code)
+}
+
+func TestEcsHealthToConsulHealth(t *testing.T) {
+	require.Equal(t, api.HealthPassing, ecsHealthToConsulHealth("HEALTHY"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth("UNHEALTHY"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth("UNKNOWN"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth(""))
+}
+
+type minimalContainerMetadata struct {
+	name      string
+	status    string
+	essential bool
+}
+
+var (
+	serviceName = "Family"
+	taskID      = "TaskId"
+	serviceID   = fmt.Sprintf("%s-%s", serviceName, taskID)
+	taskARN     = fmt.Sprintf("asdf/%s", taskID)
+)
+
+func metadataResponse(containerMetadatas []minimalContainerMetadata) string {
+	var taskMetaContainers []awsutil.ECSTaskMetaContainer
+	for _, containerMetadata := range containerMetadatas {
+		metaContainer := awsutil.ECSTaskMetaContainer{
+			Name: containerMetadata.name,
+		}
+		metaContainer.Health.Status = containerMetadata.status
+		taskMetaContainers = append(taskMetaContainers, metaContainer)
+	}
+
+	taskMeta := awsutil.ECSTaskMeta{
+		Family:     serviceName,
+		TaskARN:    taskARN,
+		Containers: taskMetaContainers,
+	}
+	json, _ := json.Marshal(taskMeta)
+	return string(json)
+}
+
+func setupServiceAndChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+		ID:   serviceID,
+		Name: serviceName,
+		Port: 1000,
+		Meta: map[string]string{
+			"task-id":  taskID,
+			"task-arn": taskARN,
+			"source":   "consul-ecs",
+		},
+	})
+	require.NoError(t, err)
+
+	for _, container := range containers {
+		if !container.essential {
+			continue
+		}
+
+		err = consulClient.Agent().CheckRegister(&api.AgentCheckRegistration{
+			Name:      makeCheckID(serviceID, container.name),
+			ServiceID: serviceID,
+			AgentServiceCheck: api.AgentServiceCheck{
+				TTL: "10000h",
+			},
```

Comment:
I'd like to include the list of checks in the `ServiceRegister` request, instead of in a separate `CheckRegister` request. So that it exactly matches the way mesh-init creates the checks.

Normally, I probably wouldn't care, and it may not matter here, but there was a recent bug I found in Consul depending on how the checks are registered ([Slack thread](https://hashicorp.slack.com/archives/C045NJD17/p1632349570455500))